src/test/java/cs361/battleships/models/BoardTest.java

```diff
@@ -186,4 +187,47 @@ public void testAtackGame(){
         g.placeShip(test,1,'A',false);
         assertEquals(true,g.attack(1,'A',false));
     }
+
+    @Test
+    public void testUderwaterSub(){
+        Game g = new Game();
+        Ship testD = new Ship("DESTROYER");
+        Ship testS = new Ship("SUB");
+        Ship testSUnerwater = new Ship("SUBB");
+        g.placeShip(testD,1,'A',false);
+        assertEquals(false,g.placeShip(testS,2,'A',false));
+        assertEquals(true,g.placeShip(testSUnerwater,2,'A',false));
+    }
+
+    @Test
+    public void testCap(){
+        Board b = new Board();
+        b.placeShip(new Ship("SUB"),2,'A',false);
+        b.placeShip(new Ship("DESTROYER"),3,'A',false);
+        b.placeShip(new Ship("MINESWEEPER"),4,'A',false);
+        b.placeShip(new Ship("BATTLESHIP"),5,'A',false);
+        b.attack(2,'A',false);
+        b.attack(3,'B',false);
+        b.attack(4,'A',false);
+        b.attack(5,'C',false);
+        // Check CC
+        var total = 0;
+        for(int i=0;i < b.getAtacks().size();i++){
+            if(b.getAtacks().get(i).getResult().equals(AtackStatus.CAPHIT)){
+                total++;
+            }
+        }
+        assertEquals(3,total);
+        b.attack(2,'A',false);
+        b.attack(3,'B',false);
+        b.attack(4,'A',false);
+        b.attack(5,'C',false);
+        total = 0;
+        for(int i=0;i < b.getAtacks().size();i++){
+            if(b.getAtacks().get(i).getResult().equals(AtackStatus.SUNK) || b.getAtacks().get(i).getResult().equals(AtackStatus.SURRENDER)){
+                total++;
+            }
+        }
+        assertEquals(4,total);
+    }
```

Comment:
Tests look good.  You check the captain's quarters functionality and the underwater functionality which is pretty much all that Is needed for the sub.